xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-api/src/main/resources/templates/extension/security/liveData/advice.vm

```diff
@@ -0,0 +1,73 @@
+## information regarding copyright ownership.
+##
+## This is free software; you can redistribute it and/or modify it
+## under the terms of the GNU Lesser General Public License as
+## published by the Free Software Foundation; either version 2.1 of
+## the License, or (at your option) any later version.
+##
+## This software is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+## Lesser General Public License for more details.
+##
+## You should have received a copy of the GNU Lesser General Public
+## License along with this software; if not, write to the Free
+## Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+## 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+## ---------------------------------------------------------------------------
+## Since 15.8RC1
+## This template renders the advice of an extension in the form of a button opening a modal with further explanations. 
+## ---------------------------------------------------------------------------
+#set ($modalId = "${extensionId}--${extensionVersion}")
+## Replace the special chars to have a valid element id.
+#set ($modalId = $modalId.replaceAll('[.:]', '-'))
+#set ($extensionFullId = "${extensionId}/${extensionVersion}")
+<button type="button" class="btn btn-default btn-xs" data-toggle="modal"
+        data-target="#$escapetool.xml($modalId)"
+        aria-controls="$escapetool.xml($modalId)">
+      <span class="sr-only">
+        $escapetool.xml($services.localization.render('extension.security.liveData.advice.modal.openButton',
+        [$extensionFullId]))
+      </span>
+  $services.icon.renderHTML('file-text')
+</button>
+<div class="modal fade security-vulnerability-detail-modal" tabindex="-1" role="dialog"
+     id="$escapetool.xml($modalId)">
+  <div class="modal-dialog modal-lg" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+        <h2 class="modal-title">
+          $escapetool.xml($services.localization.render('extension.security.liveData.advice'))
+        </h2>
+      </div>
+      <div class="modal-body">
+        $escapetool.xml($services.localization.render($adviceId))
+        #if ($backwardDependencies)
+          <ul>
+          #foreach ($backwardDependency in $backwardDependencies.entrySet())
+            <li>
+              $backwardDependency.key
+              <ul>
+                #foreach($namespace in $backwardDependency.value)##
+                  <li>
+                    #if ("$!namespace" == '')
+                      $escapetool.xml($services.localization.render('extension.security.namespaces.global'))
+                    #else
+                      $namespace
```

Comment:
Shouldn't the namespace be escaped?

Reply:
Correct. This code is now replaced with the use of `#displayExtensionNamespace($namespace)`